spec/features/restaurants/create_update_restaurant_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'rails_helper'
+
+feature 'creating and updating restaurants' do
+  context 'as an authorized user' do
+    let(:user) { FactoryGirl.create(:user) }
+    let(:restaurant) { FactoryGirl.create(:restaurant) }
+
+    before :each do
+      sign_in_as user
+    end
+
+    scenario 'I can successfully create a restaurant' do
+      visit new_restaurant_path
+
+      fill_in 'Name', with: restaurant.name
+      fill_in 'Address', with: restaurant.address
+      fill_in 'City', with: restaurant.city
+      fill_in 'Description', with: restaurant.description
+      find('#restaurant_cuisine_id').find(:xpath, 'option[1]').select_option
+      click_on 'Create Restaurant'
+
+      expect(page).to have_content('Restaurant saved successfully!')
+      expect(page).to have_content(restaurant.name)
+    end
+
+    scenario 'I can update a restaurant I create' do
+      restaurant = FactoryGirl.create(:restaurant)
+      visit restaurant_path(restaurant)
+
+      click_on 'Edit Restaurant'
+      expect(page).to have_content('Edit Restaurant')
+
+      # should be on the edit page for edited restaurant
+      fill_in 'Name', with: 'edited name'
+      fill_in 'Address', with: 'edited address'
+      click_on 'Save Changes'
+
+      # should be on the show page for edited restaurant
+      expect(page).to have_content('Restaurant saved successfully!')
+      expect(page).to have_content('edited name')
+      expect(page).to have_content('edited address')
+    end
+
+    scenario 'I can delete a restaurant I created' do
+      restaurant = FactoryGirl.create(:restaurant)
```

Comment:
Same comment as above -- is the user the same as the one who created this restaurant? If not, how would you fix this?